presto-main/src/main/java/com/facebook/presto/metadata/Metadata.java

```diff
@@ -157,6 +157,15 @@
      */
     void createTable(Session session, String catalogName, ConnectorTableMetadata tableMetadata, boolean ignoreExisting);
 
+    /**
+     * Creates a temporary table with optional partitioning requirements.
+     * Temporary table might have different default storage format, compression scheme, replication factor, etc,
+     * and gets automatically dropped when the transaction ends.
+     *
+     * This SPI is unstable and subjective to change in the future.
+     */
+    TableHandle createTemporaryTable(Session session, String catalogName, List<ColumnMetadata> columns, Optional<PartitioningMetadata> partitioningMetadata);
```

Comment:
Since the table name is not known upfront, `TableHandle` is needed to reference the temporary table.

Technically we can return the `TableHandle` from the `createTable` method as well. But there's no reason for that right now. We can do that refactor if needed.